core-common/src/main/java/org/mapstruct/Mapper.java

```diff
@@ -85,6 +85,28 @@
     String componentModel() default "default";
 
     /**
+     * Specifies the name of the implementation class. A single {@code *} wildcard can be used and will be replaced
+     * by the interface's or abstract class' name.
+     * <p>
+     * Defaults to postfixing the name with {@code Impl}
+     *
+     * @return The implementation suffix.
+     * @see #implementationPackage()
+     */
+    String implementationName() default "default";
```

Comment:
I wish we could use "<CLASS_NAME>Impl" as the default name (because that's what will be used), but I don't think we can because otherwise we could not distinguish the case where a specific `@Mapper` annotation overrides the default from `@MapperConfig` from the case where the inherited default should be used.